docs/capabilities/read_only_tenant_mode.md

```diff
@@ -0,0 +1,56 @@
+# Read Only Tenant Mode
+
+There are two distinct functionalities for "read-only" access in Dashboards. One of them is associated with roles and one is associated with tenants. Regarding the first one, the Dashboards Security plugin contains a feature of hiding all plugin navigation links except Dashboards and Visualizations when the logged-in user has a certain role. The second one is limiting Dashboards access rights via assigning a specific role to a tenant (therefore, making a tenant read-only). Due to past issues and the deprecation of the first functionality, using read-only tenants is now the recommended way to limit users' access to Dashboards.
+
+For more context, see [this group issues of problems connected with read-only roles](https://github.com/opensearch-project/security/issues/2701).
+
+## Read Only Tenant
+
+A read-only tenant looks like this:
+```
+read_only_tenant_role:
+  tenant_permissions:
+  - tenant_patterns:
+    - "human_resources"
+    allowed_actions:
+    - "kibana_all_read"
+```
+
+If the `kibana_all_read` role is assigned (without `kibana_all_write`), a given tenant provides solely read-only access.
+
+Dashboards Security plugin recognizes the selection of read-only tenant after logging in and sets the capabilities associated with write access or showing write controls to false for a variety of plugins. This can be easily checked for example by trying to re-arrange some visualizations on Dashboards. Such action will be resulting in a 403 error due to limited read-only access.
+
+## Design
+
+Whenever a plugin registers capabilities that should be limited (in other words, set to false) for read-only tenants, such capabilities should be listed in a separate capability called `hide_for_read_only` that is an array of strings, containing capabilities that are set to false whenever Dashboards Security Plugin detects a read-only tenant.
```

Comment:
Ive mentioned this in another comment but just how do the two `save` properties differ? what will happen if i do not define `save: true` and only define `hide_for_read_only: ['save'],`?